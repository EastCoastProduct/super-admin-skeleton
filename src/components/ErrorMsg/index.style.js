import { StyleSheet } from 'aphrodite/no-important';
import variables from '../../styles/variables';

const { colors } = variables;

export default StyleSheet.create({
  error: {
    backgroundColor: colors.errorBcg,
    color: colors.error,
    lineHeight: '30px',
    margin: '5px 0 15px',
    textAlign: 'center',
  },
  icon: {
    marginRight: 5,
  },
});
