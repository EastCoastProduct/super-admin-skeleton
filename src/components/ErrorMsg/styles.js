import { StyleSheet } from 'aphrodite/no-important';
import { colors } from '../../styles/variables';

export default StyleSheet.create({
  error: {
    backgroundColor: colors.errorBcg,
    color: colors.error,
    margin: '10px 0',
    padding: '6.5px 10px',
    textAlign: 'center',
  },
  icon: {
    marginRight: 5,
  },
});
