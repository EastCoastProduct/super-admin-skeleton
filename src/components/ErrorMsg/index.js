import React, { PropTypes } from 'react';
import { css } from 'aphrodite/no-important';
import styles from './index.style';

const ErrorMsg = ({ children, style }) =>
  <p className={css(styles.error, style && style)}>
    <i className={`${css(styles.icon)} fa fa-exclamation-triangle`} />
    {children}
  </p>;

ErrorMsg.propTypes = {
  children: PropTypes.string.isRequired,
  style: PropTypes.object,
};

export default ErrorMsg;
