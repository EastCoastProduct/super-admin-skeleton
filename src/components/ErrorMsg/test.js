import React from 'react';
import { shallow } from 'enzyme';
import { StyleSheet } from 'aphrodite/no-important';
import ErrorMsg from './';

describe('ErrorMsg component snapshot', () => {
  it('renders basic required data without htmlFor', () => {
    const wrapper = shallow(<ErrorMsg>Something went wrong.</ErrorMsg>);

    expect(wrapper).toMatchSnapshot();
  });

  it('renders basic required data with htmlFor', () => {
    const wrapper = shallow(
      <ErrorMsg htmlFor="Login-email">Something went wrong.</ErrorMsg>
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('renders with additional styles', () => {
    const styles = StyleSheet.create({
      additional: {
        margin: 0,
        padding: 0,
      },
    });
    const wrapper = shallow(
      <ErrorMsg className={styles.additional}>Something went wrong.</ErrorMsg>
    );

    expect(wrapper).toMatchSnapshot();
  });
});
