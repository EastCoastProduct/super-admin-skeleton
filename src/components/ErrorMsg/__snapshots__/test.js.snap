exports[`ErrorMsg component snapshot renders basic required data with htmlFor 1`] = `
<label
  className="error_1pg0y0x"
  htmlFor="Login-email">
  <i
    className="icon_1w16ibx fa fa-exclamation-triangle" />
  Something went wrong.
</label>
`;

exports[`ErrorMsg component snapshot renders basic required data without htmlFor 1`] = `
<p
  className="error_1pg0y0x">
  <i
    className="icon_1w16ibx fa fa-exclamation-triangle" />
  Something went wrong.
</p>
`;

exports[`ErrorMsg component snapshot renders with additional styles 1`] = `
<p
  className="error_1pg0y0x-o_O-additional_1433df3">
  <i
    className="icon_1w16ibx fa fa-exclamation-triangle" />
  Something went wrong.
</p>
`;
